File: server/models/AdminLog.js
const mongoose = require('mongoose');

const AdminLogSchema = new mongoose.Schema({
    adminId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    adminEmail: {
        type: String,
        required: true
    },
    action: {
        type: String,
        required: true
    },
    targetId: {
        type: String
    },
    targetModel: {
        type: String
    },
    details: {
        type: mongoose.Schema.Types.Mixed
    }
}, {
    timestamps: true
});

module.exports = mongoose.model('AdminLog', AdminLogSchema);

File: lib/features/admin/domain/entities/admin_log.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'admin_log.freezed.dart';
part 'admin_log.g.dart';

@freezed
class AdminLog with _$AdminLog {
  const factory AdminLog({
    String? id,
    required String adminId,
    required String adminEmail,
    required String action,
    String? targetId,
    String? targetModel,
    String? details,
    DateTime? createdAt,
  }) = _AdminLog;

  factory AdminLog.fromJson(Map<String, dynamic> json) => _$AdminLogFromJson(json);
}
